internal/groups/service.go

```diff
@@ -696,7 +694,7 @@ func (svc service) changeGroupStatus(ctx context.Context, token string, group gr
 		return groups.Group{}, err
 	}
 	if dbGroup.Status == group.Status {
-		return groups.Group{}, mgclients.ErrStatusAlreadyAssigned
+		return groups.Group{}, errors.ErrStatusAlreadyAssigned
```

Comment:
@rodneyosodo, No need to change the following comments in this PR. 

We need to have some kind of error layers. 

Idea is to prevent or avoid sending of Reposistory errors in API response 

One of the way which I have in my mind to implement this idea is, reposistory should always return erorrs from pacakge `repoerr`
service should return errors from package `svcerr`, 

In API error encoder, We need to check for api errors and service errors and send only api errors and service errors in response and don't want to send repo errors in response. 

Example: 
for GET request of `things/<thingID>`
In response If entity is not found, then i should get error message of `svcerr.ErrNotFound`, not from repository layer, error return by postgres library.

In logs all the errors should be present. So if there we can get the more error details from service logs.

Reply:
This can be done with something like this https://github.com/absmach/magistrala-ui/pull/240 whereby all errors from the repository layer must be wrapped with the service layer unless specified otherwise. But it is highly intrusive.

Reply:
Created a issue [here](https://github.com/absmach/magistrala/issues/2117) for this comment